clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/DeleteManifestIT.java

```diff
@@ -0,0 +1,347 @@
+          baseUrl(), b, account1Ns, String.format("replicaSet %s-v%03d", rsn, i));
+    }
+    List<Map<String, Object>> rb =
+        buildDynamicRequestBody(
+            String.format("%s %s", rs, rdn), true, String.format("%s %s", rs, rsn), c, rs);
+    List<String> on = KubeTestUtils.sendOperation(baseUrl(), rb, account1Ns);
+    String ss =
+        kubeCluster.execKubectl(
+            String.format("-n %s get %s %s --ignore-not-found", account1Ns, rs, rdn));
+    assertTrue(ss.isBlank() && on.size() == 1 && on.get(0).equals(String.format("%s %s", rs, rdn)));
+  }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given a deployment manifest with 3 replicas outside of Spinnaker\n"
+          + "When sending a delete manifest operation without cascading option enable\n"
+          + "Then just the deployment should be remove at once\n===")
+  @Test
+  public void shouldDeleteWithoutCascading() throws IOException, InterruptedException {
+    String d = "deployment";
+    String dn = "myapp";
+    Map<String, Object> dm =
+        KubeTestUtils.loadYaml("classpath:manifests/deployment.yml")
+            .withValue("metadata.name", dn)
+            .withValue("spec.replicas", 3)
+            .asMap();
+    kubeCluster.execKubectl("-n " + account1Ns + " apply -f -", dm);
+    kubeCluster.execKubectl(
+        String.format(
+            "wait %s -n %s %s --for condition=Available=True --timeout=600s", d, account1Ns, dn));
+    List<Map<String, Object>> rb = buildStaticRequestBody(String.format("%s %s", d, dn), false);
+    List<String> on = KubeTestUtils.sendOperation(baseUrl(), rb, account1Ns);
+    String ps = kubeCluster.execKubectl(String.format("-n %s get pods -l=app=%s", account1Ns, dn));
+    assertTrue(
+        on.size() == 1
+            && on.get(0).equals(String.format("%s %s", d, dn))
+            && ps.contains("Running"));
+  }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given a NOT existing static deployment manifest\n"
+          + "When sending a delete manifest request\n"
+          + "Then it should return any deleted deployment\n===")
+  @Test
+  public void shouldNotDeleteStaticTarget() throws InterruptedException {
+    List<Map<String, Object>> rb = buildStaticRequestBody("deployment notExists", true);
+    List<String> on = KubeTestUtils.sendOperation(baseUrl(), rb, account1Ns);
+    assertEquals(0, on.size());
+  }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given a NOT existing dynamic replicaSet manifest\n"
+          + "When sending a delete manifest operation using smallest dynamic target criteria\n"
+          + "Then it gets a 404 while fetching the manifest\n===")
+  @Test
+  public void shouldNotFoundDynamicTarget() throws InterruptedException {
+    String rs = "replicaSet";
+    String c = "smallest";
+    String rsn = String.format("not-exists-%s-test", c);
+    String rdn = String.format("%s-v000", rsn);
+    assertThrows(
+        AssertionFailedError.class,
```

Comment:
This test looks looks like it can be simplified to just sending the `GET /manifests` request and expecting a 404. Asserting that it throws an assert exception is like having a double negated `if` condition.